source/production/F0.Minesweeper.Components/Services/VersionInfo.cs

```diff
@@ -0,0 +1,30 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using NuGet.Versioning;
+
+namespace F0.Minesweeper.Components.Services
+{
+	public sealed class VersionInfo
+	{
+		public VersionInfo(Assembly? assembly)
+		{
+			if (assembly is not null)
+			{
+				ProductName = assembly.GetCustomAttribute<AssemblyTitleAttribute>()?.Title;
+				string? version = assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion;
+				ProductVersion = SemanticVersion.Parse(version);
```

Comment:
If `version` is `null` then `SemanticVersion.Parse(version)` will throw an `ArgumentException`, so we can either handle the null case or we assume that `assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()` must return a value, so we can just use `string version`.

Reply:
fixing via `TryParse`